contracts/chain-adapters/Arbitrum_CustomGasToken_L2_Forwarder.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { CircleCCTPAdapter, CircleDomainIds } from "../libraries/CircleCCTPAdapter.sol";
+import { ITokenMessenger as ICCTPTokenMessenger } from "../external/interfaces/CCTPInterfaces.sol";
+import { ArbitrumCustomGasTokenInbox, ArbitrumL1ERC20GatewayLike } from "../interfaces/ArbitrumBridgeInterfaces.sol";
+import { ForwarderBase } from "./ForwarderBase.sol";
+import { Arbitrum_CustomGasToken_AdapterBase, FunderInterface } from "./Arbitrum_CustomGasToken_AdapterBase.sol";
+
+/**
+ * @notice Contract containing logic to send messages from Arbitrum to an AVM L3.
+ * @dev This contract is very similar to Arbitrum_CustomGasToken_Adapter. It is meant to bridge
+ * tokens and send messages over a bridge which uses a custom gas token, except this contract assumes
```

Comment:
I would also note that this contract is designed to be controlled by a cross domain admin which should be the HubPool

Reply:
https://github.com/across-protocol/contracts/pull/610/commits/05dfc6eefae26f129ab68223a011766894ec144b